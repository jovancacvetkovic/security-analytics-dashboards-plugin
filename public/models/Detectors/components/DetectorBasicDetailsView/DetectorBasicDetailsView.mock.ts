import { mockDetector } from '../../../Interfaces.mock';

export default {
  detector: mockDetector,
  rulesCanFold: true,
  enabled_time: 1,
  last_update_time: 1,
  onEditClicked: () => jest.fn(),
};
