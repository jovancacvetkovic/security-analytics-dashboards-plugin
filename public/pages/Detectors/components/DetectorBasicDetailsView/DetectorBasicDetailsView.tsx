/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import { EuiButton, EuiSpacer, EuiLink, EuiIcon } from '@elastic/eui';
import React from 'react';
import { ContentPanel } from '../../../../components/ContentPanel';
import { createTextDetailsGroup, parseSchedule } from '../../../../utils/helpers';
import moment from 'moment';
import { DEFAULT_EMPTY_DATA } from '../../../../utils/constants';
import { Detector } from '../../../../../types';

export interface DetectorBasicDetailsViewProps {
  detector: Detector;
  dashboardId?: string;
  rulesCanFold?: boolean;
  enabled_time?: number;
  last_update_time?: number;
  onEditClicked: () => void;
  isEditable: boolean;
}

export const DetectorBasicDetailsView: React.FC<DetectorBasicDetailsViewProps> = ({
  detector,
  enabled_time,
  last_update_time,
  rulesCanFold,
  children,
  dashboardId,
  onEditClicked,
  isEditable = true,
}) => {
  const { name, detector_type, inputs, schedule } = detector;
  const detectorSchedule = parseSchedule(schedule);
  const createdAt = enabled_time ? moment(enabled_time).format('YYYY-MM-DDTHH:mm') : undefined;
  const lastUpdated = last_update_time
    ? moment(last_update_time).format('YYYY-MM-DDTHH:mm')
    : undefined;
  const firstTextDetailsGroupEntries = [
    { label: 'Detector name', content: name },
    { label: 'Log type', content: detector_type.toLowerCase() },
    { label: 'Data source', content: inputs[0].detector_input.indices[0] },
    {
      label: 'Detector dashboard',
      content: (dashboardId ? (
        <EuiLink onClick={() => window.open(`dashboards#/view/${dashboardId}`, '_blank')}>
          {`${name} summary`}
          <EuiIcon type={'popout'} />
        </EuiLink>
      ) : (
        'Not available for this log type'
      )) as any,
    },
  ];

  return (
    <ContentPanel
      title={'Detector details'}
      actions={
        isEditable
          ? [
              <EuiButton onClick={onEditClicked} data-test-subj={'edit-detector-basic-details'}>
                Edit
              </EuiButton>,
            ]
          : null
      }
    >
      <EuiSpacer size={'l'} />
      {createTextDetailsGroup(firstTextDetailsGroupEntries, 4)}
      {createTextDetailsGroup(
        [
          { label: 'Description', content: inputs[0].detector_input.description },
          { label: 'Detector schedule', content: detectorSchedule },
          { label: 'Created at', content: createdAt || DEFAULT_EMPTY_DATA },
          { label: 'Last updated time', content: lastUpdated || DEFAULT_EMPTY_DATA },
        ],
        4
      )}
      {rulesCanFold ? children : null}
    </ContentPanel>
  );
};
