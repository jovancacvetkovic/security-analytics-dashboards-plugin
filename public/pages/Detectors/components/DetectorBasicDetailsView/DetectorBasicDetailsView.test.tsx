/*
 * Copyright OpenSearch Contributors
 * SPDX-License-Identifier: Apache-2.0
 */

import React from 'react';
import { render } from '@testing-library/react';
import { DetectorBasicDetailsView } from './DetectorBasicDetailsView';
import props from '../../../../../test/mocks/Detectors/components/DetectorBasicDetailsView/DetectorBasicDetailsView.mock';
import { expect } from '@jest/globals';

describe('<DetectorBasicDetailsView /> spec', () => {
  it('renders the component', () => {
    const view = render(<DetectorBasicDetailsView {...props} />);
    expect(view).toMatchSnapshot();
  });
});
