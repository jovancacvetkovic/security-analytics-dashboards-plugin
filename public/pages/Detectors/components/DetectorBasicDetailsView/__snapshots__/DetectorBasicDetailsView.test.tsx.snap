// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DetectorBasicDetailsView /> spec renders the component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
        style="padding-left: 0px; padding-right: 0px;"
      >
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
          style="padding: 0px 10px;"
        >
          <div
            class="euiFlexItem"
          >
            <h3
              class="euiTitle euiTitle--medium"
            >
              Detector details
            </h3>
          </div>
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
          >
            <div
              class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
            >
              <div
                class="euiFlexItem"
              >
                <button
                  class="euiButton euiButton--primary"
                  data-test-subj="edit-detector-basic-details"
                  type="button"
                >
                  <span
                    class="euiButtonContent euiButton__content"
                  >
                    <span
                      class="euiButton__text"
                    >
                      Edit
                    </span>
                  </span>
                </button>
              </div>
            </div>
          </div>
        </div>
        <hr
          class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
        />
        <div
          style="padding: 0px 10px;"
        >
          <div
            class="euiSpacer euiSpacer--l"
          />
          <div
            class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
          >
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
              style="min-width: 25%;"
            >
              <div
                class="euiFormRow"
                id="some_html_id-row"
              >
                <div
                  class="euiFormRow__labelWrapper"
                >
                  <label
                    class="euiFormLabel euiFormRow__label"
                    for="some_html_id"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      <div
                        class="euiTextColor euiTextColor--subdued"
                      >
                        Detector name
                      </div>
                    </div>
                  </label>
                </div>
                <div
                  class="euiFormRow__fieldWrapper"
                >
                  <div
                    class="euiText euiText--medium"
                    data-test-subj="text-details-group-content-detector-name"
                    id="some_html_id"
                  >
                    detector_name
                  </div>
                </div>
              </div>
            </div>
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
              style="min-width: 25%;"
            >
              <div
                class="euiFormRow"
                id="some_html_id-row"
              >
                <div
                  class="euiFormRow__labelWrapper"
                >
                  <label
                    class="euiFormLabel euiFormRow__label"
                    for="some_html_id"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      <div
                        class="euiTextColor euiTextColor--subdued"
                      >
                        Log type
                      </div>
                    </div>
                  </label>
                </div>
                <div
                  class="euiFormRow__fieldWrapper"
                >
                  <div
                    class="euiText euiText--medium"
                    data-test-subj="text-details-group-content-log-type"
                    id="some_html_id"
                  >
                    detector_type
                  </div>
                </div>
              </div>
            </div>
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
              style="min-width: 25%;"
            >
              <div
                class="euiFormRow"
                id="some_html_id-row"
              >
                <div
                  class="euiFormRow__labelWrapper"
                >
                  <label
                    class="euiFormLabel euiFormRow__label"
                    for="some_html_id"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      <div
                        class="euiTextColor euiTextColor--subdued"
                      >
                        Data source
                      </div>
                    </div>
                  </label>
                </div>
                <div
                  class="euiFormRow__fieldWrapper"
                >
                  <div
                    class="euiText euiText--medium"
                    data-test-subj="text-details-group-content-data-source"
                    id="some_html_id"
                  >
                    .windows
                  </div>
                </div>
              </div>
            </div>
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
              style="min-width: 25%;"
            >
              <div
                class="euiFormRow"
                id="some_html_id-row"
              >
                <div
                  class="euiFormRow__labelWrapper"
                >
                  <label
                    class="euiFormLabel euiFormRow__label"
                    for="some_html_id"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      <div
                        class="euiTextColor euiTextColor--subdued"
                      >
                        Detector dashboard
                      </div>
                    </div>
                  </label>
                </div>
                <div
                  class="euiFormRow__fieldWrapper"
                >
                  <div
                    class="euiText euiText--medium"
                    data-test-subj="text-details-group-content-detector-dashboard"
                    id="some_html_id"
                  >
                    Not available for this log type
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="euiSpacer euiSpacer--xl"
          />
          <div
            class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
          >
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
              style="min-width: 25%;"
            >
              <div
                class="euiFormRow"
                id="some_html_id-row"
              >
                <div
                  class="euiFormRow__labelWrapper"
                >
                  <label
                    class="euiFormLabel euiFormRow__label"
                    for="some_html_id"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      <div
                        class="euiTextColor euiTextColor--subdued"
                      >
                        Description
                      </div>
                    </div>
                  </label>
                </div>
                <div
                  class="euiFormRow__fieldWrapper"
                >
                  <div
                    class="euiText euiText--medium"
                    data-test-subj="text-details-group-content-description"
                    id="some_html_id"
                  >
                    detectorDescription
                  </div>
                </div>
              </div>
            </div>
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
              style="min-width: 25%;"
            >
              <div
                class="euiFormRow"
                id="some_html_id-row"
              >
                <div
                  class="euiFormRow__labelWrapper"
                >
                  <label
                    class="euiFormLabel euiFormRow__label"
                    for="some_html_id"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      <div
                        class="euiTextColor euiTextColor--subdued"
                      >
                        Detector schedule
                      </div>
                    </div>
                  </label>
                </div>
                <div
                  class="euiFormRow__fieldWrapper"
                >
                  <div
                    class="euiText euiText--medium"
                    data-test-subj="text-details-group-content-detector-schedule"
                    id="some_html_id"
                  >
                    Every 1 minute
                  </div>
                </div>
              </div>
            </div>
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
              style="min-width: 25%;"
            >
              <div
                class="euiFormRow"
                id="some_html_id-row"
              >
                <div
                  class="euiFormRow__labelWrapper"
                >
                  <label
                    class="euiFormLabel euiFormRow__label"
                    for="some_html_id"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      <div
                        class="euiTextColor euiTextColor--subdued"
                      >
                        Created at
                      </div>
                    </div>
                  </label>
                </div>
                <div
                  class="euiFormRow__fieldWrapper"
                >
                  <div
                    class="euiText euiText--medium"
                    data-test-subj="text-details-group-content-created-at"
                    id="some_html_id"
                  >
                    2023-01-25T10:05
                  </div>
                </div>
              </div>
            </div>
            <div
              class="euiFlexItem euiFlexItem--flexGrowZero"
              style="min-width: 25%;"
            >
              <div
                class="euiFormRow"
                id="some_html_id-row"
              >
                <div
                  class="euiFormRow__labelWrapper"
                >
                  <label
                    class="euiFormLabel euiFormRow__label"
                    for="some_html_id"
                  >
                    <div
                      class="euiText euiText--medium"
                    >
                      <div
                        class="euiTextColor euiTextColor--subdued"
                      >
                        Last updated time
                      </div>
                    </div>
                  </label>
                </div>
                <div
                  class="euiFormRow__fieldWrapper"
                >
                  <div
                    class="euiText euiText--medium"
                    data-test-subj="text-details-group-content-last-updated-time"
                    id="some_html_id"
                  >
                    2023-01-25T10:05
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="euiSpacer euiSpacer--xl"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="euiPanel euiPanel--paddingMedium euiPanel--borderRadiusMedium euiPanel--plain euiPanel--hasShadow"
      style="padding-left: 0px; padding-right: 0px;"
    >
      <div
        class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
        style="padding: 0px 10px;"
      >
        <div
          class="euiFlexItem"
        >
          <h3
            class="euiTitle euiTitle--medium"
          >
            Detector details
          </h3>
        </div>
        <div
          class="euiFlexItem euiFlexItem--flexGrowZero"
        >
          <div
            class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--alignItemsCenter euiFlexGroup--justifyContentSpaceBetween euiFlexGroup--directionRow euiFlexGroup--responsive"
          >
            <div
              class="euiFlexItem"
            >
              <button
                class="euiButton euiButton--primary"
                data-test-subj="edit-detector-basic-details"
                type="button"
              >
                <span
                  class="euiButtonContent euiButton__content"
                >
                  <span
                    class="euiButton__text"
                  >
                    Edit
                  </span>
                </span>
              </button>
            </div>
          </div>
        </div>
      </div>
      <hr
        class="euiHorizontalRule euiHorizontalRule--full euiHorizontalRule--marginXSmall"
      />
      <div
        style="padding: 0px 10px;"
      >
        <div
          class="euiSpacer euiSpacer--l"
        />
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
            style="min-width: 25%;"
          >
            <div
              class="euiFormRow"
              id="some_html_id-row"
            >
              <div
                class="euiFormRow__labelWrapper"
              >
                <label
                  class="euiFormLabel euiFormRow__label"
                  for="some_html_id"
                >
                  <div
                    class="euiText euiText--medium"
                  >
                    <div
                      class="euiTextColor euiTextColor--subdued"
                    >
                      Detector name
                    </div>
                  </div>
                </label>
              </div>
              <div
                class="euiFormRow__fieldWrapper"
              >
                <div
                  class="euiText euiText--medium"
                  data-test-subj="text-details-group-content-detector-name"
                  id="some_html_id"
                >
                  detector_name
                </div>
              </div>
            </div>
          </div>
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
            style="min-width: 25%;"
          >
            <div
              class="euiFormRow"
              id="some_html_id-row"
            >
              <div
                class="euiFormRow__labelWrapper"
              >
                <label
                  class="euiFormLabel euiFormRow__label"
                  for="some_html_id"
                >
                  <div
                    class="euiText euiText--medium"
                  >
                    <div
                      class="euiTextColor euiTextColor--subdued"
                    >
                      Log type
                    </div>
                  </div>
                </label>
              </div>
              <div
                class="euiFormRow__fieldWrapper"
              >
                <div
                  class="euiText euiText--medium"
                  data-test-subj="text-details-group-content-log-type"
                  id="some_html_id"
                >
                  detector_type
                </div>
              </div>
            </div>
          </div>
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
            style="min-width: 25%;"
          >
            <div
              class="euiFormRow"
              id="some_html_id-row"
            >
              <div
                class="euiFormRow__labelWrapper"
              >
                <label
                  class="euiFormLabel euiFormRow__label"
                  for="some_html_id"
                >
                  <div
                    class="euiText euiText--medium"
                  >
                    <div
                      class="euiTextColor euiTextColor--subdued"
                    >
                      Data source
                    </div>
                  </div>
                </label>
              </div>
              <div
                class="euiFormRow__fieldWrapper"
              >
                <div
                  class="euiText euiText--medium"
                  data-test-subj="text-details-group-content-data-source"
                  id="some_html_id"
                >
                  .windows
                </div>
              </div>
            </div>
          </div>
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
            style="min-width: 25%;"
          >
            <div
              class="euiFormRow"
              id="some_html_id-row"
            >
              <div
                class="euiFormRow__labelWrapper"
              >
                <label
                  class="euiFormLabel euiFormRow__label"
                  for="some_html_id"
                >
                  <div
                    class="euiText euiText--medium"
                  >
                    <div
                      class="euiTextColor euiTextColor--subdued"
                    >
                      Detector dashboard
                    </div>
                  </div>
                </label>
              </div>
              <div
                class="euiFormRow__fieldWrapper"
              >
                <div
                  class="euiText euiText--medium"
                  data-test-subj="text-details-group-content-detector-dashboard"
                  id="some_html_id"
                >
                  Not available for this log type
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="euiSpacer euiSpacer--xl"
        />
        <div
          class="euiFlexGroup euiFlexGroup--gutterLarge euiFlexGroup--directionRow euiFlexGroup--responsive"
        >
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
            style="min-width: 25%;"
          >
            <div
              class="euiFormRow"
              id="some_html_id-row"
            >
              <div
                class="euiFormRow__labelWrapper"
              >
                <label
                  class="euiFormLabel euiFormRow__label"
                  for="some_html_id"
                >
                  <div
                    class="euiText euiText--medium"
                  >
                    <div
                      class="euiTextColor euiTextColor--subdued"
                    >
                      Description
                    </div>
                  </div>
                </label>
              </div>
              <div
                class="euiFormRow__fieldWrapper"
              >
                <div
                  class="euiText euiText--medium"
                  data-test-subj="text-details-group-content-description"
                  id="some_html_id"
                >
                  detectorDescription
                </div>
              </div>
            </div>
          </div>
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
            style="min-width: 25%;"
          >
            <div
              class="euiFormRow"
              id="some_html_id-row"
            >
              <div
                class="euiFormRow__labelWrapper"
              >
                <label
                  class="euiFormLabel euiFormRow__label"
                  for="some_html_id"
                >
                  <div
                    class="euiText euiText--medium"
                  >
                    <div
                      class="euiTextColor euiTextColor--subdued"
                    >
                      Detector schedule
                    </div>
                  </div>
                </label>
              </div>
              <div
                class="euiFormRow__fieldWrapper"
              >
                <div
                  class="euiText euiText--medium"
                  data-test-subj="text-details-group-content-detector-schedule"
                  id="some_html_id"
                >
                  Every 1 minute
                </div>
              </div>
            </div>
          </div>
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
            style="min-width: 25%;"
          >
            <div
              class="euiFormRow"
              id="some_html_id-row"
            >
              <div
                class="euiFormRow__labelWrapper"
              >
                <label
                  class="euiFormLabel euiFormRow__label"
                  for="some_html_id"
                >
                  <div
                    class="euiText euiText--medium"
                  >
                    <div
                      class="euiTextColor euiTextColor--subdued"
                    >
                      Created at
                    </div>
                  </div>
                </label>
              </div>
              <div
                class="euiFormRow__fieldWrapper"
              >
                <div
                  class="euiText euiText--medium"
                  data-test-subj="text-details-group-content-created-at"
                  id="some_html_id"
                >
                  2023-01-25T10:05
                </div>
              </div>
            </div>
          </div>
          <div
            class="euiFlexItem euiFlexItem--flexGrowZero"
            style="min-width: 25%;"
          >
            <div
              class="euiFormRow"
              id="some_html_id-row"
            >
              <div
                class="euiFormRow__labelWrapper"
              >
                <label
                  class="euiFormLabel euiFormRow__label"
                  for="some_html_id"
                >
                  <div
                    class="euiText euiText--medium"
                  >
                    <div
                      class="euiTextColor euiTextColor--subdued"
                    >
                      Last updated time
                    </div>
                  </div>
                </label>
              </div>
              <div
                class="euiFormRow__fieldWrapper"
              >
                <div
                  class="euiText euiText--medium"
                  data-test-subj="text-details-group-content-last-updated-time"
                  id="some_html_id"
                >
                  2023-01-25T10:05
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          class="euiSpacer euiSpacer--xl"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
